src/CommandingOfficers/CommanderStrong.java

```diff
@@ -1,40 +1,163 @@
+    if( null == highCapacityUnitModels )
+    {
+      UnitModel hcApc = new APCModel();
+      UnitModel hcLander = new LanderModel();
+      UnitModel hcCopter = new TCopterModel();
+      highCapacityUnitModels = new HashMap<UnitModel.UnitEnum, UnitModel>();
+      highCapacityUnitModels.put(hcApc.type, hcApc);
+      highCapacityUnitModels.put(hcLander.type, hcLander);
+      highCapacityUnitModels.put(hcCopter.type, hcCopter);
+      for( UnitModel um : highCapacityUnitModels.values() )
+      {
+        um.holdingCapacity += 1;
+      }
+      // TODO: other non-transport types?
+    }
+
+    // Strong allows infantry to be built from any production building.
+    UnitModel infModel = getUnitModel(UnitModel.UnitEnum.INFANTRY);
+    UnitProductionModifier upm = new UnitProductionModifier(TerrainType.AIRPORT, infModel);
+    upm.addProductionPair(TerrainType.SEAPORT, infModel);
+    upm.addProductionPair(TerrainType.HEADQUARTERS, infModel);
+    upm.apply(this); // Passive ability, so don't add it to the COModifier list; just apply it and forget it.
+
+    // Set Strong up with a base damage buff and long-range APCs. These COModifiers are
+    // not added to the modifiers collection so they will not be reverted.
     COModifier strongMod = new CODamageModifier(20); // Give us a nice base power boost.
     strongMod.apply(this);
 
     COMovementModifier moveMod = new COMovementModifier();
     moveMod.addApplicableUnitType(UnitModel.UnitEnum.APC);
     moveMod.apply(this);
+
+    addCommanderAbility(new StrongArmAbility(this));
+    addCommanderAbility(new MobilizeAbility(this));
   }
 
-  public void doAbilityMinor()
+  /**
+   * StrongArm grants Strong a firepower bonus, additional mobility,
+   * and the ability to build Bazooka units from all production buildings and the HQ.
+   */
+  private static class StrongArmAbility extends CommanderAbility
   {
-    COModifier strongerMod = new CODamageModifier(20); // Another 20% with the minor ability active.
-    strongerMod.apply(this);
-    modifiers.add(strongerMod); // Add to the list so the modifier can be reverted next turn.
+    private static final String STRONGARM_NAME = "Strongarm";
+    private static final int STRONGARM_COST = 10;
+    private static final int STRONGARM_BUFF = 20;
+
+    COModifier damageMod = null;
+
+    StrongArmAbility(Commander commander)
+    {
+      super(commander, STRONGARM_NAME, STRONGARM_COST);
+
+      damageMod = new CODamageModifier(STRONGARM_BUFF);
+    }
+
+    @Override
+    protected void perform(GameInstance game)
+    {
+      myCommander.addCOModifier(damageMod);
+
+      // Make bazookas buildable from all production buildings.
+      UnitModel mechModel = myCommander.getUnitModel(UnitModel.UnitEnum.MECH);
+      UnitProductionModifier upm = new UnitProductionModifier(TerrainType.AIRPORT, mechModel);
+      upm.addProductionPair(TerrainType.SEAPORT, mechModel);
+      upm.addProductionPair(TerrainType.HEADQUARTERS, mechModel);
```

Comment:
No infantry from HQ?

Reply:
He gets infantry from HQ by default. Which is maybe a bit much, so maybe we should change that, and put it here instead. Thoughts?

Reply:
Yeah, I think an extra infantry-production building might be a bit much in the early game. I'm going to push that to his abilities only.